R/tbl_likert.R

```diff
@@ -0,0 +1,253 @@
+#' Likert Summary
+#'
+#' `r lifecycle::badge("experimental")`\cr
+#' Create a table of ordered categorical variables in a wide format.
+#'
+#' @inheritParams tbl_summary
+#' @param statistic ([`formula-list-selector`][syntax])\cr
+#'   Used to specify the summary statistics for each variable.
+#'   The default is `everything() ~ "{n} ({p}%)"`.
+#' @param digits ([`formula-list-selector`][syntax])\cr
+#'  Specifies how summary statistics are rounded. Values may be either integer(s)
+#'   or function(s). If not specified, default formatting is assigned
+#'   via `assign_summary_digits()`.
+#' @param sort (`string`)\cr
+#'   indicates whether levels of variables should be placed in
+#'   ascending order (the default) or descending.
+#'
+#' @return a 'tbl_likert' gtsummary table
+#' @export
+#'
+#' @examples
+#' levels <- c("Strongly Disagree", "Disagree", "Agree", "Strongly Agree")
+#' df_likert <- data.frame(
+#'   recommend_friend = sample(levels, size = 20, replace = TRUE) |> factor(levels = levels),
+#'   regret_purchase = sample(levels, size = 20, replace = TRUE) |> factor(levels = levels)
+#' )
+#'
+#' # Example 1 ----------------------------------
+#' tbl_likert_ex1 <-
+#'   df_likert |>
+#'   tbl_likert() |>
+#'   add_n()
+#' tbl_likert_ex1
+#'
+#' # Example 2 ----------------------------------
+#' # Add continuous summary of the likert scores
```

Comment:
Thanks for the example. This is fine for me. With `tbl_wide_summary()`, not necessary to include `add_continuous_stat()`